### PR TITLE
KAFKA-10759 Add ARM build stage

### PR DESCRIPTION
**Pull request by mumrah**
Copy of https://github.com/apache/kafka/pull/9872, but opened by a committer so Jenkins will actually apply the Jenkinsfile changes

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -158,6 +158,23 @@ pipeline {
...
+        stage('ARM') {
+          agent { label 'arm4' }
+          options {
+            timeout(time: 8, unit: 'HOURS') 
+            timestamps()
+          }
+          environment {
+            SCALA_VERSION=2.12
+          }
+          steps {
```
Comment (xiao-penglei) [inline Jenkinsfile:171]: Would it be better to add error processing, for example "catchError(buildResult: 'SUCCESS', stageResult: 'FAILURE')" .
In this case, the arm CI fails, which does not affect the CI build result of the entire Kafka.

**Review by junrao — APPROVED** [review]
@mumrah : Thanks for the PR. LGTM assuming the ARM test passes.

**Comment by mumrah** [comment]
Looks like one of the tests got hung up and the build timed out:

```
16:04:09  SocketServerTest > testStagedListenerShutdownWhenConnectionQueueIsFull() STARTED
16:04:09  kafka.network.SocketServerTest.testStagedListenerShutdownWhenConnectionQueueIsFull() failed, log available in /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR-9992/core/build/reports/testOutput/kafka.network.SocketServerTest.testStagedListenerShutdownWhenConnectionQueueIsFull().test.stdout
16:04:09  
16:04:09  SocketServerTest > testStagedListenerShutdownWhenConnectionQueueIsFull() FAILED
16:04:09      java.net.BindException: Cannot assign requested address (Bind failed)
16:04:09          at java.net.PlainSocketImpl.socketBind(Native Method)
16:04:09          at java.net.AbstractPlainSocketImpl.bind(AbstractPlainSocketImpl.java:387)
16:04:09          at java.net.Socket.bind(Socket.java:662)
16:04:09          at java.net.Socket.<init>(Socket.java:451)
16:04:09          at java.net.Socket.<init>(Socket.java:304)
16:04:09          at kafka.network.SocketServerTest.connect(SocketServerTest.scala:155)
16:04:09          at kafka.network.SocketServerTest.testStagedListenerShutdownWhenConnectionQueueIsFull(SocketServerTest.scala:355)
16:04:09  
16:04:09  SocketServerTest > testStagedListenerStartup() STARTED
17:20:50  Sending interrupt signal to process
17:21:04  Terminated
17:21:05  script returned exit code 143
```

**Comment by xiao-penglei** [comment]
> Looks like one of the tests got hung up and the build timed out:
> 
> ```
> 16:04:09  SocketServerTest > testStagedListenerShutdownWhenConnectionQueueIsFull() STARTED
> 16:04:09  kafka.network.SocketServerTest.testStagedListenerShutdownWhenConnectionQueueIsFull() failed, log available in /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR-9992/core/build/reports/testOutput/kafka.network.SocketServerTest.testStagedListenerShutdownWhenConnectionQueueIsFull().test.stdout
> 16:04:09  
> 16:04:09  SocketServerTest > testStagedListenerShutdownWhenConnectionQueueIsFull() FAILED
> 16:04:09      java.net.BindException: Cannot assign requested address (Bind failed)
> 16:04:09          at java.net.PlainSocketImpl.socketBind(Native Method)
> 16:04:09          at java.net.AbstractPlainSocketImpl.bind(AbstractPlainSocketImpl.java:387)
> 16:04:09          at java.net.Socket.bind(Socket.java:662)
> 16:04:09          at java.net.Socket.<init>(Socket.java:451)
> 16:04:09          at java.net.Socket.<init>(Socket.java:304)
> 16:04:09          at kafka.network.SocketServerTest.connect(SocketServerTest.scala:155)
> 16:04:09          at kafka.network.SocketServerTest.testStagedListenerShutdownWhenConnectionQueueIsFull(SocketServerTest.scala:355)
> 16:04:09  
> 16:04:09  SocketServerTest > testStagedListenerStartup() STARTED
> 17:20:50  Sending interrupt signal to process
> 17:21:04  Terminated
> 17:21:05  script returned exit code 143
> ```

@mumrah I tried to execute the test case on the jenkins arm4 node with the jenkins account in the same environment. The result is as follows:
> ```
>> Configure project :
>Building project 'core' with Scala version 2.12.12
>Building project 'streams-scala' with Scala version 2.12.12
>
>> Task :core:compileScala
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/main/scala/kafka/coordinator/group/MemberMetadata.scala:31: private object MemberMetadata in package >group is never used
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/main/scala/kafka/network/RequestChannel.scala:155: parameter value nn in method body is never used
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR-9992/core/src/main/scala/kafka/utils/NotNothing.scala:40: >parameter value n in method notNothingEvidence is never used
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/main/scala/kafka/utils/json/DecodeJson.scala:88: higher-kinded type should be enabled
>by making the implicit value scala.language.higherKinds visible.
>This can be achieved by adding the import clause 'import scala.language.higherKinds'
>or by setting the compiler option -language:higherKinds.
>See the Scaladoc for value scala.language.higherKinds for a discussion
>why the feature should be explicitly enabled.
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/main/scala/kafka/utils/json/DecodeJson.scala:94: higher-kinded type should be enabled
>by making the implicit value scala.language.higherKinds visible.
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/main/scala/kafka/admin/ConfigCommand.scala:354: method alterConfigs in trait Admin is deprecated
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/main/scala/kafka/security/authorizer/AuthorizerUtils.scala:40: class AuthorizerWrapper in package authorizer is >deprecated (since Since 2.5): Use kafka.security.authorizer.AclAuthorizer
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR-9992/core/src/main/scala/kafka/server/KafkaApis.scala:620: >constructor ProduceResponse in class ProduceResponse is deprecated
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR-9992/core/src/main/scala/kafka/server/KafkaApis.scala:627: >constructor ProduceResponse in class ProduceResponse is deprecated
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/main/scala/kafka/tools/ConsoleConsumer.scala:629: method checksum in class ConsumerRecord is deprecated
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/main/scala/kafka/tools/MirrorMaker.scala:195: class BaseConsumerRecord in package consumer is >deprecated(since 0.11.0.0): This class has been deprecated and will be removed in a future release. Please use >org.apache.kafka.clients.consumer.ConsumerRecord instead.
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/main/scala/kafka/tools/MirrorMaker.scala:196: class BaseConsumerRecord in package consumer is >deprecated(since 0.11.0.0): This class has been deprecated and will be removed in a future release. Please use >org.apache.kafka.clients.consumer.ConsumerRecord instead.
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/main/scala/kafka/tools/MirrorMaker.scala:418: class BaseConsumerRecord in package consumer is >deprecated(since 0.11.0.0): This class has been deprecated and will be removed in a future release. Please use >org.apache.kafka.clients.consumer.ConsumerRecord instead.
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/main/scala/kafka/tools/MirrorMaker.scala:423: class BaseConsumerRecord in package consumer is >deprecated(since 0.11.0.0): This class has been deprecated and will be removed in a future release. Please use >org.apache.kafka.clients.consumer.ConsumerRecord instead.
>14 warnings found
>
>> Task :core:compileTestScala
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/test/scala/unit/kafka/server/BaseRequestTest.scala:90: parameter value nn in method receive is never used
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/test/scala/integration/kafka/api/AdminClientWithPoliciesIntegrationTest.scala:122: method alterConfigs in trait >Admin is deprecated
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/test/scala/integration/kafka/api/AdminClientWithPoliciesIntegrationTest.scala:150: method alterConfigs in trait >Admin is deprecated
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/test/scala/integration/kafka/api/AuthorizerIntegrationTest.scala:165: method responses in class >ProduceResponse is deprecated
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/test/scala/integration/kafka/api/AuthorizerIntegrationTest.scala:1000: method poll in class KafkaConsumer is >deprecated
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/test/scala/integration/kafka/api/AuthorizerIntegrationTest.scala:1048: method poll in class KafkaConsumer is >deprecated
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/test/scala/integration/kafka/api/AuthorizerIntegrationTest.scala:1077: method poll in class KafkaConsumer is >deprecated
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/test/scala/integration/kafka/api/BaseProducerSendTest.scala:126: method checksum in class RecordMetadata is >deprecated
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/test/scala/integration/kafka/api/ConsumerBounceTest.scala:102: method poll in class KafkaConsumer is >deprecated
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/test/scala/integration/kafka/api/ConsumerBounceTest.scala:392: method poll in class KafkaConsumer is >deprecated
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/test/scala/integration/kafka/api/PlaintextAdminIntegrationTest.scala:2186: method alterConfigs in trait Admin is >deprecated
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/test/scala/integration/kafka/api/PlaintextAdminIntegrationTest.scala:2241: method alterConfigs in trait Admin is >deprecated
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/test/scala/integration/kafka/api/PlaintextAdminIntegrationTest.scala:2273: method alterConfigs in trait Admin is >deprecated
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/test/scala/integration/kafka/api/PlaintextAdminIntegrationTest.scala:2314: method alterConfigs in trait Admin is >deprecated
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/test/scala/integration/kafka/api/PlaintextAdminIntegrationTest.scala:2342: method alterConfigs in trait Admin is >deprecated
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/test/scala/integration/kafka/api/SaslSslAdminIntegrationTest.scala:50: class LegacyAuthorizationAdmin in class >SaslSslAdminIntegrationTest is deprecated (since Since 2.5): Use kafka.security.authorizer.AclAuthorizer
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/test/scala/integration/kafka/network/DynamicConnectionQuotaTest.scala:343: method responses in class >ProduceResponse is deprecated
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/test/scala/integration/kafka/network/DynamicConnectionQuotaTest.scala:344: method responses in class >ProduceResponse is deprecated
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/test/scala/integration/kafka/server/DynamicBrokerReconfigurationTest.scala:1371: method alterConfigs in trait >Admin is deprecated
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/test/scala/integration/kafka/server/DynamicBrokerReconfigurationTest.scala:1390: method alterConfigs in trait >Admin is deprecated
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/test/scala/integration/kafka/server/DynamicBrokerReconfigurationTest.scala:1407: method alterConfigs in trait >Admin is deprecated
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/test/scala/unit/kafka/integration/UncleanLeaderElectionTest.scala:349: method alterConfigs in trait Admin is >deprecated
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/test/scala/unit/kafka/security/authorizer/AuthorizerWrapperTest.scala:40: class AuthorizerWrapper in package >authorizer is deprecated (since Since 2.5): Use kafka.security.authorizer.AclAuthorizer
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/test/scala/unit/kafka/security/authorizer/AuthorizerWrapperTest.scala:40: class SimpleAclAuthorizer in package >auth is deprecated (since Since 2.4): Use kafka.security.authorizer.AclAuthorizer
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/test/scala/unit/kafka/security/authorizer/AuthorizerWrapperTest.scala:42: class AuthorizerWrapper in package >authorizer is deprecated (since Since 2.5): Use kafka.security.authorizer.AclAuthorizer
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/test/scala/unit/kafka/security/authorizer/AuthorizerWrapperTest.scala:42: class SimpleAclAuthorizer in package >auth is deprecated (since Since 2.4): Use kafka.security.authorizer.AclAuthorizer
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/test/scala/unit/kafka/security/authorizer/AuthorizerWrapperTest.scala:57: object SimpleAclAuthorizer in package >auth is deprecated (since Since 2.4): Use kafka.security.authorizer.AclAuthorizer
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/test/scala/unit/kafka/server/EdgeCaseRequestTest.scala:157: method responses in class ProduceResponse is >deprecated
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/test/scala/unit/kafka/server/EdgeCaseRequestTest.scala:159: method responses in class ProduceResponse is >deprecated
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/test/scala/unit/kafka/server/KafkaApisTest.scala:1263: method responses in class ProduceResponse is deprecated
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/test/scala/unit/kafka/server/KafkaApisTest.scala:1264: method responses in class ProduceResponse is deprecated
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/test/scala/unit/kafka/server/ProduceRequestTest.scala:64: method responses in class ProduceResponse is >deprecated
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/test/scala/unit/kafka/server/ProduceRequestTest.scala:65: method responses in class ProduceResponse is >deprecated
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/test/scala/unit/kafka/server/ProduceRequestTest.scala:113: method responses in class ProduceResponse is >deprecated
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/test/scala/unit/kafka/server/ProduceRequestTest.scala:155: method responses in class ProduceResponse is >deprecated
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/test/scala/unit/kafka/server/ProduceRequestTest.scala:156: method responses in class ProduceResponse is >deprecated
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/test/scala/unit/kafka/server/ProduceRequestTest.scala:188: method responses in class ProduceResponse is >deprecated
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/test/scala/unit/kafka/server/ProduceRequestTest.scala:189: method responses in class ProduceResponse is >deprecated
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/test/scala/unit/kafka/server/ProduceRequestTest.scala:227: method responses in class ProduceResponse is >deprecated
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/test/scala/unit/kafka/server/ProduceRequestTest.scala:237: method responses in class ProduceResponse is >deprecated
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/test/scala/unit/kafka/tools/MirrorMakerTest.scala:35: class BaseConsumerRecord in package consumer is >deprecated (since 0.11.0.0): This class has been deprecated and will be removed in a future release. Please use >org.apache.kafka.clients.consumer.ConsumerRecord instead.
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/test/scala/unit/kafka/tools/MirrorMakerTest.scala:50: class BaseConsumerRecord in package consumer is >deprecated (since 0.11.0.0): This class has been deprecated and will be removed in a future release. Please use >org.apache.kafka.clients.consumer.ConsumerRecord instead.
>[Warn] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR->9992/core/src/test/scala/unit/kafka/tools/MirrorMakerTest.scala:67: class BaseConsumerRecord in package consumer is >deprecated (since 0.11.0.0): This class has been deprecated and will be removed in a future release. Please use >org.apache.kafka.clients.consumer.ConsumerRecord instead.
>43 warnings found
>
>> Task :core:test
>
>SocketServerTest > testStagedListenerShutdownWhenConnectionQueueIsFull() PASSED
>
>Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
>Use '--warning-mode all' to show the individual deprecation warnings.
>See https://docs.gradle.org/6.8.1/userguide/command_line_interface.html#sec:command_line_warnings
>
>BUILD SUCCESSFUL in 8m 12s
>25 actionable tasks: 21 executed, 4 up-to-date
>
> ```

**Comment by xiao-penglei** [comment]
> > Looks like one of the tests got hung up and the build timed out:
> > ```
> > 16:04:09  SocketServerTest > testStagedListenerShutdownWhenConnectionQueueIsFull() STARTED
> > 16:04:09  kafka.network.SocketServerTest.testStagedListenerShutdownWhenConnectionQueueIsFull() failed, log available in /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR-9992/core/build/reports/testOutput/kafka.network.SocketServerTest.testStagedListenerShutdownWhenConnectionQueueIsFull().test.stdout
> > 16:04:09  
> > 16:04:09  SocketServerTest > testStagedListenerShutdownWhenConnectionQueueIsFull() FAILED
> > 16:04:09      java.net.BindException: Cannot assign requested address (Bind failed)
> > 16:04:09          at java.net.PlainSocketImpl.socketBind(Native Method)
> > 16:04:09          at java.net.AbstractPlainSocketImpl.bind(AbstractPlainSocketImpl.java:387)
> > 16:04:09          at java.net.Socket.bind(Socket.java:662)
> > 16:04:09          at java.net.Socket.<init>(Socket.java:451)
> > 16:04:09          at java.net.Socket.<init>(Socket.java:304)
> > 16:04:09          at kafka.network.SocketServerTest.connect(SocketServerTest.scala:155)
> > 16:04:09          at kafka.network.SocketServerTest.testStagedListenerShutdownWhenConnectionQueueIsFull(SocketServerTest.scala:355)
> > 16:04:09  
> > 16:04:09  SocketServerTest > testStagedListenerStartup() STARTED
> > 17:20:50  Sending interrupt signal to process
> > 17:21:04  Terminated
> > 17:21:05  script returned exit code 143
> > ```
> 
> @mumrah I tried to execute the test case on the jenkins arm4 node with the jenkins account in the same environment. The result is as follows:
> 
> > ```
> > root@monero:/home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR-9992# su jenkins
> > $ ./gradlew core:test --tests kafka.network.SocketServerTest.testStagedListenerShutdownWhenConnectionQueueIsFull
> > 
> > > Configure project :
> > Building project 'core' with Scala version 2.13.4
> > Building project 'streams-scala' with Scala version 2.13.4
> > 
> > > Task :core:test
> > 
> > SocketServerTest > testStagedListenerShutdownWhenConnectionQueueIsFull() PASSED
> > 
> > Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
> > Use '--warning-mode all' to show the individual deprecation warnings.
> > See https://docs.gradle.org/6.8.1/userguide/command_line_interface.html#sec:command_line_warnings
> > 
> > BUILD SUCCESSFUL in 10m 5s
> > 25 actionable tasks: 21 executed, 4 up-to-date
> > $ ./gradlew core:test --tests kafka.network.SocketServerTest.testStagedListenerShutdownWhenConnectionQueueIsFull
> > 
> > > Configure project :
> > Building project 'core' with Scala version 2.13.4
> > Building project 'streams-scala' with Scala version 2.13.4
> > 
> > Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
> > Use '--warning-mode all' to show the individual deprecation warnings.
> > See https://docs.gradle.org/6.8.1/userguide/command_line_interface.html#sec:command_line_warnings
> > 
> > BUILD SUCCESSFUL in 4s
> > 25 actionable tasks: 2 executed, 23 up-to-date
> > ```

@mumrah  I will continue to locate the causeof the problem

**Comment by xiao-penglei** [comment]
> Looks like one of the tests got hung up and the build timed out:
> 
> ```
> 16:04:09  SocketServerTest > testStagedListenerShutdownWhenConnectionQueueIsFull() STARTED
> 16:04:09  kafka.network.SocketServerTest.testStagedListenerShutdownWhenConnectionQueueIsFull() failed, log available in /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR-9992/core/build/reports/testOutput/kafka.network.SocketServerTest.testStagedListenerShutdownWhenConnectionQueueIsFull().test.stdout
> 16:04:09  
> 16:04:09  SocketServerTest > testStagedListenerShutdownWhenConnectionQueueIsFull() FAILED
> 16:04:09      java.net.BindException: Cannot assign requested address (Bind failed)
> 16:04:09          at java.net.PlainSocketImpl.socketBind(Native Method)
> 16:04:09          at java.net.AbstractPlainSocketImpl.bind(AbstractPlainSocketImpl.java:387)
> 16:04:09          at java.net.Socket.bind(Socket.java:662)
> 16:04:09          at java.net.Socket.<init>(Socket.java:451)
> 16:04:09          at java.net.Socket.<init>(Socket.java:304)
> 16:04:09          at kafka.network.SocketServerTest.connect(SocketServerTest.scala:155)
> 16:04:09          at kafka.network.SocketServerTest.testStagedListenerShutdownWhenConnectionQueueIsFull(SocketServerTest.scala:355)
> 16:04:09  
> 16:04:09  SocketServerTest > testStagedListenerStartup() STARTED
> 17:20:50  Sending interrupt signal to process
> 17:21:04  Terminated
> 17:21:05  script returned exit code 143
> ```

@mumrah   I also tried to execute the unitTest and integrationTest on the jenkins arm4 node with the jenkins account in the same environment. The result is as follows:

> ```
>$ ./gradlew -PscalaVersion=2.12 unitTest integrationTest --profile --no-daemon --continue ->PtestLoggingEvents=started,passed,skipped,failed -PignoreFailures=true -PmaxParallelForks=2 -PmaxTestRetries=1 ->PmaxTestRetryFailures=5
>...
>...
>Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
>Use '--warning-mode all' to show the individual deprecation warnings.
>See https://docs.gradle.org/6.8.1/userguide/command_line_interface.html#sec:command_line_warnings
>
>BUILD SUCCESSFUL in 2h 12m 6s
>170 actionable tasks: 149 executed, 21 up-to-date
>
>See the profiling report at: file:///home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR-9992/build/reports/profile/profile->2021-02-02-08-31-54.html
>A fine-grained performance profile is available: use the --scan option.
>
>
> Try again without clean
>
>$ ./gradlew -PscalaVersion=2.12 unitTest integrationTest --profile --no-daemon --continue ->PtestLoggingEvents=started,passed,skipped,failed -PignoreFailures=true -PmaxParallelForks=2 -PmaxTestRetries=1 ->PmaxTestRetryFailures=5
>To honour the JVM settings for this build a single-use Daemon process will be forked. See >https://docs.gradle.org/6.8.1/userguide/gradle_daemon.html#sec:disabling_the_daemon.
>Daemon will be stopped at the end of the build 
>
>> Configure project :
>Building project 'core' with Scala version 2.12.12
>Building project 'streams-scala' with Scala version 2.12.12
>
>Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
>Use '--warning-mode all' to show the individual deprecation warnings.
>See https://docs.gradle.org/6.8.1/userguide/command_line_interface.html#sec:command_line_warnings
>
>BUILD SUCCESSFUL in 31s
>170 actionable tasks: 3 executed, 167 up-to-date
>
>See the profiling report at: file:///home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR-9992/build/reports/profile/profile->2021-02-02-10-50-04.html
>A fine-grained performance profile is available: use the --scan option.
>
> ```

**Comment by xiao-penglei** [comment]
> Looks like one of the tests got hung up and the build timed out:
> 
> ```
> 16:04:09  SocketServerTest > testStagedListenerShutdownWhenConnectionQueueIsFull() STARTED
> 16:04:09  kafka.network.SocketServerTest.testStagedListenerShutdownWhenConnectionQueueIsFull() failed, log available in /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR-9992/core/build/reports/testOutput/kafka.network.SocketServerTest.testStagedListenerShutdownWhenConnectionQueueIsFull().test.stdout
> 16:04:09  
> 16:04:09  SocketServerTest > testStagedListenerShutdownWhenConnectionQueueIsFull() FAILED
> 16:04:09      java.net.BindException: Cannot assign requested address (Bind failed)
> 16:04:09          at java.net.PlainSocketImpl.socketBind(Native Method)
> 16:04:09          at java.net.AbstractPlainSocketImpl.bind(AbstractPlainSocketImpl.java:387)
> 16:04:09          at java.net.Socket.bind(Socket.java:662)
> 16:04:09          at java.net.Socket.<init>(Socket.java:451)
> 16:04:09          at java.net.Socket.<init>(Socket.java:304)
> 16:04:09          at kafka.network.SocketServerTest.connect(SocketServerTest.scala:155)
> 16:04:09          at kafka.network.SocketServerTest.testStagedListenerShutdownWhenConnectionQueueIsFull(SocketServerTest.scala:355)
> 16:04:09  
> 16:04:09  SocketServerTest > testStagedListenerStartup() STARTED
> 17:20:50  Sending interrupt signal to process
> 17:21:04  Terminated
> 17:21:05  script returned exit code 143
> ```

@mumrah I did find from [#3](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-9992/3/)  [#4](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-9992/4/) [#5](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-9992/5/) [#6](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-9992/6/)  that every failure was due to this test case 'SocketServerTest > testStagedListenerShutdownWhenConnectionQueueIsFull' . I tried to reproduce the problem on arm4 node, but unfortunately I didn't get the desired result. Because the testOutput directory has been deleted, I cannot see the error output of this case.  Can you help me re-trigger the CI? I'll spend the time analyzing testOutput. Thanks

**Comment by xiao-penglei** [comment]
@mumrah [#8](https://ci-builds.apache.org/blue/organizations/jenkins/Kafka%2Fkafka-pr/detail/PR-9992/8/pipeline) .It looks good. If there are failed test cases on the ARM platform after the integration, I will take time to investigate. Thanks

**Comment by xiao-penglei** [comment]
All unit test cases are ok in arm build stage now. Can we merge it to ensure that the arm build stage can be executed for each PR? @mumrah @ijuma @junrao

**Pull request merged**